testhelpers/docker.go

```diff
@@ -108,16 +106,26 @@ func DockerVolumeExists(t *testing.T, volumeName string) bool {
 }
 
 // FIXME: re-work this function to exec the docker cli, or convert other docker helpers to using the client library.
-func PushImage(dockerCli dockercli.CommonAPIClient, ref string, auth string) error {
-	rc, err := dockerCli.ImagePush(context.Background(), ref, dockertypes.ImagePushOptions{RegistryAuth: auth})
-	if err != nil {
-		return errors.Wrap(err, "pushing image")
+func PushImage(ref string, auth string) error {
+	// Filter out any existing DOCKER_CONFIG from the environment
+	filteredEnv := []string{}
+	for _, envVar := range os.Environ() {
+		if !strings.HasPrefix(envVar, "DOCKER_CONFIG=") {
+			filteredEnv = append(filteredEnv, envVar)
+		}
 	}
+	// Append the new DOCKER_CONFIG to the filtered environment
+	newEnv := append(filteredEnv, "DOCKER_CONFIG="+auth)
```

Comment:
Ah, I think `auth` here is a JSON encoded string, whereas docker expects the location of a directory containing a config.json file. If you look further up to where `auth` is constructed it should be possible to find that directory.